src/action.cpp

```diff
@@ -1189,5 +1189,10 @@ std::optional<tripoint> choose_adjacent_highlight( const tripoint &pos, const st
         g->add_draw_callback( hilite_cb );
     }
 
-    return choose_adjacent( pos, message, allow_vertical );
+    const std::optional<tripoint> chosen = choose_adjacent( pos, message, allow_vertical );
+    if( std::find( valid.begin(), valid.end(), chosen ) != valid.end() ) {
+        return chosen;
+    }
```

Comment:
You should probably check if the current usages of `choose_adjacent_highlight` rely on the non-highlighted tiles being selectable.

Reply:
Looking at the code choose_adjacent could arleady return std::nullopt so this should not cause crashes. And I fail to find a case where we would highlight the valid tiles but still want you to be able to chose an invalid one.